app/src/main/java/org/ea/sqrl/activites/identity/IdentitySettingsActivity.java

```diff
@@ -110,25 +97,83 @@ public void onResume() {
+     */
+    private void validateInputFields() {
+        mInputFieldsValid = true;
+
+        String text = txtQuickPassLength.getText().toString();
+        txtQuickPassLength.setError(null);
+
+        if (text.equals("")) {
+            setInputFieldError(txtQuickPassLength, R.string.error_field_may_not_be_empty);
+        } else if (!Utils.isNumeric(text)) {
+            setInputFieldError(txtQuickPassLength, R.string.settings_hint_length_not_number);
+        } else if (getIntValue(txtQuickPassLength) > 255) {
+            setInputFieldError(txtQuickPassLength, R.string.error_value_too_large_0_to_255);
+        }
+
+        text = txtPwdVerifySecs.getText().toString();
+        txtPwdVerifySecs.setError(null);
+
+        if (text.equals("")) {
+            setInputFieldError(txtPwdVerifySecs, R.string.error_field_may_not_be_empty);
+        } else if (!Utils.isNumeric(text)) {
+            setInputFieldError(txtPwdVerifySecs, R.string.settings_password_verify_not_number);
+        } else if (getIntValue(txtPwdVerifySecs) > 255) {
+            setInputFieldError(txtPwdVerifySecs, R.string.error_value_too_large_0_to_255);
+        }
+
+        text = txtQuickPassTimeout.getText().toString();
+        txtQuickPassTimeout.setError(null);
+
+        if (text.equals("")) {
+            setInputFieldError(txtQuickPassTimeout, R.string.error_field_may_not_be_empty);
+        } else if (!Utils.isNumeric(text)) {
+            setInputFieldError(txtQuickPassTimeout, R.string.settings_idle_timeout_not_number);
+        } else if (getIntValue(txtQuickPassTimeout) > 65535) {
+            setInputFieldError(txtQuickPassTimeout, R.string.error_value_too_large_0_to_65535);
+        }
+
+        btnSettingsSave.setEnabled(mInputFieldsValid);
+    }
+
+    private void setInputFieldError(EditText textField, int errorStringId) {
+        textField.setError(getResources().getString(errorStringId));
+        mInputFieldsValid = false;
     }
 
     private void update() {
         final SQRLStorage storage = SQRLStorage.getInstance(IdentitySettingsActivity.this.getApplicationContext());
 
-        txtSettingsQuickPassLength.setText(Integer.toString(storage.getHintLength()));
-        txtSettingsPasswordVerify.setText(Integer.toString(storage.getPasswordVerify()));
-        txtSettingsQuickPassTimeout.setText(Integer.toString(storage.getIdleTimeout()));
-        cbSettingsSQRLOnly.setChecked(storage.isSQRLOnly());
-        cbSettingsNoBypass.setChecked(storage.isNoByPass());
+        txtQuickPassLength.setText(Integer.toString(storage.getHintLength()));
+        txtPwdVerifySecs.setText(Integer.toString(storage.getPasswordVerify()));
+        txtQuickPassTimeout.setText(Integer.toString(storage.getIdleTimeout()));
+        cbSQRLOnly.setChecked(storage.isSQRLOnly());
+        cbNoBypass.setChecked(storage.isNoByPass());
     }
 
-    public int getIntValue(EditText txt, int errorMessage) {
+    public int getIntValue(EditText txt) {
```

Comment:
Is -1 a valid number? It is below 65535 at least.

Reply:
This method always returned -1 on error, I haven't changed anything there. I've just removed the call for showing the error message. Or am I missing your point?

Reply:
If you're talking about lower bounds checking on the input fields, that would need to happen in `validateInputFields()`. And yes, I've thought about it too. I guess we should only allow positive integers. Right? If you agree, I'll modify my code to accomodate that.

Reply:
Yes, that was what I was thinking about. Might have expressed it a bit bad.